sentinelhub/download/session.py

```diff
@@ -5,63 +5,91 @@
 import json
 import logging
 import time
+import warnings
+from typing import Any, Dict, Optional, Union
 
 from oauthlib.oauth2 import BackendApplicationClient
 from requests_oauthlib import OAuth2Session
 
 from ..config import SHConfig
 from ..download.handlers import fail_user_errors, retry_temporary_errors
 from ..download.request import DownloadRequest
+from ..exceptions import SHUserWarning
 
 LOGGER = logging.getLogger(__name__)
 
+JsonDict = Dict[str, Any]
```

Comment:
The biggest benefit of having a name for it that it is easier to understand what kind of data this is and see all the places where it is used. Using `Dict[str, Any]` doesn't tell that much about the data itself and it is not that obvious if the types in different methods are the same by coincidence or because they all originate from the same source.

For now, I only put `JsonDict` into this module but probably we could define all commonly used types at one place in the package and then reuse them everywhere.